ecs_logging/_stdlib.py

```diff
@@ -42,9 +43,24 @@ class StdlibFormatter(logging.Formatter):
     }
     converter = time.gmtime
 
-    def __init__(self):
-        # type: () -> None
+    def __init__(self, include_exc_info=False, stack_trace_limit=0):
+        """Initialize the ECS formatter.
+
+        Parameters
+        ----------
+        include_exc_info : bool
```

Comment:
I agree most users will probably want exceptions; however, that would change the behavior from the current version. While it's still call-compatible (in that a default value is provided), a caller that upgrades without changing their code will start getting exception information populated where it wasn't before. Is that an issue for you?

Reply:
I'd say that since we're sill pre v1, ~10 downloads per day, and that most users will want this enabled anyways that this is fine to "break" behavior here.

In the future when things solidify and are out of the early stages we'll definitely not introduce default behavior changes.

Reply:
So here is the next question: what's the intent of `WANTED_ATTRS` here? Is the expectation that a user would subclass `StdlibFormatter` and override `WANTED_ATTRS` to specify which ECS keys to populate or not? Or do you envision another way for a user to specify which keys they want populated (where the default implementation just adds everything it knows how to add)? The exception info kind of breaks the current model for it, because 1 property on a `LogRecord` (`exc_info`) is mapping to 3 fields in an ECS record. If you are willing to contemplate something a little more general that might break backwards-compatibility here, then we can probably get rid of the `include_exc_info` parameter altogether in favor of a new mechanism, and just retain `stack_trace_limit` and default it to `None` (as the `limit` parameter on `traceback.*` does).

Reply:
So the brief history of the "StdlibFormatter" is that it was lifted from Curator and not modified much so `WANTED_ATTRS` being a public class attribute is a "unneeded appendage" and something that should change to be internal.

What I'm imagining for sub-classing your own Formatter would be to subclass and implement `format_to_ecs()`, that way if you wanted to add your own data to each record after calling `super().format_to_ecs()` you can.

Unfortunately there's no precedent for adding "contexts" like you can in structlog to stdlib loggers so the only per-record influence you'd have is using `extra={...}` which is currently merged into the ECS doc too.